src/GitHub.TeamFoundation.14/Base/TeamExplorerServiceHolder.cs

```diff
@@ -142,8 +152,14 @@ void UpdateActiveRepo()
             var repo = gitService?.ActiveRepositories.FirstOrDefault();
 
             if (!Equals(repo, ActiveRepo))
-                // so annoying that this is on the wrong thread
-                syncContext.Post(r => ActiveRepo = r as ILocalRepositoryModel, repo);
+            {
+                // Fire property change events on Main thread
```

Comment:
I'm glad you spotted this. It would have changed the behavior from the previous `SynchronizationContext.Post(...)`! I missed the fact that `Post` is asynchranous.